# Orobanchaceae_O2K
Scripts used in the comparative physiological investigation of mitochondrial genome function in the parasitic Orobanchaceae

Citation: Cai, Liming, Robert K. Jansen, and Justin C. Havird. "Altered mitochondrial respiration is associated with loss of nuclear-encoded OXPHOS genes in parasitic broomrapes." bioRxiv (2025): 2025-02.

Subfolder contents:

**1_Angiosperm_Nmt_survey**

Scripts used to fish N-mt genes, clean paralogs, align sequence, and generate preliminary gene trees 
	
**2_Species_tree_inference**

Further remove paralogs using the phylogeny-based Yang and Smith pipeline, generate one-to-one ortholog for species tree inference 

**3_Gene_loss_verification**

BUSCO evaluation of data quality, phylogentic ANOVA analysis, Fisher's exact test for gene loss

**4_Molecular_evolution**

N-terminal targeting peptides identification and removal; HyPhy commands for RELAX analysis
	
**5_O2K_statistic_analysis**

R script to analyze O2K data
